### DISCUSSION TITLE
Correct num_time_steps in Time Domain Simulation?

### DISCUSSION THREAD

**Discussion by usmandroid** [NVlabs/sionna#117] [Q&A] [answered]
Hello!

I am following this tutorial [MIMO OFDM Transmissions over the CDL Channel Model](https://nvlabs.github.io/sionna/examples/MIMO_OFDM_Transmissions_over_CDL.html#CDL-Channel-Model) and I have discovered a discrepancy and an issue with correct value for num_time_steps when performing Time Domain OFDM simulation.

First discrepancy:
At input [10] `a, tau = cdl(batch_size=32, num_time_steps=rg.num_ofdm_symbols, sampling_frequency=1/rg.ofdm_symbol_duration)`

At input [16] `a, tau = cdl(batch_size=2, num_time_steps=rg.num_time_samples+l_tot-1, sampling_frequency=rg.bandwidth)`

**which of them is correct [10] or [16]?**

Second issue:

I believe case [16] is correct, assuming it is, then `rg.num_time_samples` represents  `rg.fft_size*rg.num_ofdm_symbols` rather than `(rg.fft_size + rg.cyclic_prefix_length)*rg.num_ofdm_symbols` .

If we use `num_time_steps=rg.fft_size*rg.num_ofdm_symbols` as an argument **the CIR is shorter than the total OFDM signal length** leading to a loss of OFDM symbols. When I tried this I initially had 18 symbols and after applying time channel I was left with only 13 symbols assuming fft_size=64 and cyclic_prefix_length=16.

**Comment by jhoydis** [comment, marked as answer]
Hi,

I believe that there is no issue in the code.

[10] is for simulations in the frequency domain. That is, we need one CIR sample for each OFDM symbol.

[16] is for simulations in the time domain. Each OFDM symbol consists of ``rg.fft_size + rg.cyclic_prefix_length`` samples. The CIR has length ``l_tot`` so that the output signal after the convolution has length ``(rg.fft_size + rg.cyclic_prefix_length)*rg.num_ofdm_symbols + l_tot - 1``.

In your example, the number of time steps is too low because you forgot to include the cyclic prefix.

**Reply by usmandroid** [reply]
Hi Mr. Hoydis,

Thank you for the clarification. Indeed the correct value for `num_time_steps` should be `(rg.fft_size + rg.cyclic_prefix_length)*rg.num_ofdm_symbols + l_tot - 1` since `rg.num_time_samples` represents `(rg.fft_size + rg.cyclic_prefix_length)*rg.num_ofdm_symbols`. 

For Frequency Domain simulation 
- `cyclic_prefix_length = 0`  assuming perfect synchronization, no ISI/ICI therefore `num_time_steps=rg.num_ofdm_symbols`

For Time Domain simulation 
- `cyclic_prefix_length > l_tot`
